islandora_audio.module

```diff
@@ -210,6 +210,19 @@ function islandora_audio_islandora_sp_audiocmodel_islandora_ingest_steps() {
  * Implements hook_CMODEL_PID_islandora_derivative().
  */
 function islandora_audio_islandora_sp_audiocmodel_islandora_derivative() {
+  if (variable_get('islandora_audio_defer_derivatives_on_ingest', FALSE)) {
+    return array(
+      array(
+        'source_dsid' => NULL,
+        'destination_dsid' => 'TN',
+        'weight' => 1,
+        'function' => array(
+          'islandora_audio_create_thumbnail',
+        ),
+        'file' => drupal_get_path('module', 'islandora_audio') . '/includes/derivatives.inc',
+      ),
+    );
+  }
```

Comment:
All and all this PR is fine with me, to avoid duplication of code. I would suggest instead of doing
```
if (variable_get(...
return array
}
return array
```
You store the array and use your ```if (variable_get``` to either add or not add the PROXY_MP3 derivative to that array.
If I am making any sense :confused: